nodestream_plugin_neptune/query_executor.py

```diff
@@ -0,0 +1,118 @@
+from logging import getLogger
+from typing import Iterable
+from nodestream.model import IngestionHook, Node, RelationshipWithNodes, TimeToLiveConfiguration
+from nodestream.schema.indexes import FieldIndex, KeyIndex
+from nodestream.databases.query_executor import (
+    OperationOnNodeIdentity,
+    OperationOnRelationshipIdentity,
+    QueryExecutor,
+)
+from .ingest_query_builder import NeptuneDBIngestQueryBuilder
+from .query import Query, QueryBatch
+from aiobotocore.session import get_session
+import json
+import asyncio
+import math
+
+
+class NeptuneQueryExecutor(QueryExecutor):
+    def __init__(
+        self,
+        region,
+        host,
+        ingest_query_builder: NeptuneDBIngestQueryBuilder,
+        async_partitions = 50
+    ) -> None:
+        self.session = get_session()
+        self.region = region
+        self.host = host
+        self.ingest_query_builder = ingest_query_builder
+        self.logger = getLogger(self.__class__.__name__)
+        self.async_partitions = async_partitions
+    async def upsert_nodes_in_bulk_with_same_operation(
+        self, operation: OperationOnNodeIdentity, nodes: Iterable[Node]
+    ):
+        batched_query = (
+            self.ingest_query_builder.generate_batch_update_node_operation_batch(
+                operation, nodes
+            )
+        )
+        await self.execute(batched_query.as_query())
+
+    async def upsert_relationships_in_bulk_of_same_operation(
+        self,
+        shape: OperationOnRelationshipIdentity,
+        relationships: Iterable[RelationshipWithNodes],
+    ):
+        queries = (
+            self.ingest_query_builder.generate_batch_update_relationship_query_batch(
+                shape, relationships
+            )
+        )
+        await self.execute(queries.as_query())
+
+    async def upsert_key_index(self, index: KeyIndex):
+        # Not needed for Neptune
+        pass
+
+    async def upsert_field_index(self, index: FieldIndex):
+        # Not needed for Neptune
+        pass
+
+    async def perform_ttl_op(self, config: TimeToLiveConfiguration):
+        # TODO: Implement as appropriate
+        pass
+
+    async def execute_hook(self, hook: IngestionHook):
+        # TODO: Implement as appropriate
+        pass
+
```

Comment:
This one is simple. `IngestionHook` implements `as_cypher_query_and_parameters` which should be easy to work with which you should be able to pass to `execute`.